algo-core/scheduler_utils.py

```diff
@@ -0,0 +1,148 @@
+import argparse
+import json
+import sys
+from pathlib import Path
+import jsonschema
+import pandas as pd
+
+
+# Input filenames:
+filename_onboarding = "onboarding_agents.txt"
+filename_mentors = "mentors.txt"
+filename_new = "new_agents.txt"
+input_folder = Path()
+
+def get_project_root() -> Path:
+    return Path(__file__).parent.parent
+
+
+def parse_json_input():
+    """Read, validate and return json input."""
+    # Production (read input from command line):
+    global input_folder
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-i", "--input", help="Scheduler input JSON file path", required=True
+    )
+    args = parser.parse_args()
+    input_filename = args.input.strip()
+
+    # Testing (define input directly):
+    # input_filename = "support-shift-scheduler-input.json"
+
+    # Load and validate JSON input:
+    input_json = json.load(open(input_filename))
+    path_to_schema = Path(get_project_root(), "lib", "schemas" , "support-shift-scheduler-input.schema.json")
+    input_json_schema = json.load(open(path_to_schema))
+    try:
+        jsonschema.validate(input_json, input_json_schema)
+    except jsonschema.exceptions.ValidationError as err:
+        print("Input JSON validation error", err)
+        sys.exit(1)
+
+    input_folder = get_project_root(), "logs" , f"{input_json['options']['startMondayDate']}_{input_json['options']['modelName']}"
```

Comment:
Does this work??

If we're separating pieces by commas, I'd expect the `Path` constructor
```suggestion
    input_folder = Path(get_project_root(), "logs" , f"{input_json['options']['startMondayDate']}_{input_json['options']['modelName']}")
```